src/tests/test_magic.py

```diff
@@ -1296,3 +1297,170 @@ def test_interact_and_missing_ipywidgets_installed(ip):
+            (
+                "'jupysql' not found from pyproject.toml. Please check the following "
+                "section names: 'invalid_section'."
+            ),
+        ),
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = true
+autocommit = true
+""",
+            ValueError,
+            "Duplicate key found : 'autocommit'",
+        ),
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = True
+""",
+            ValueError,
+            (
+                "Invalid value 'True' in 'autocommit = True'. "
+                "Valid boolean values: true, false"
+            ),
+        ),
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = invalid
+""",
+            ValueError,
+            (
+                "Invalid value 'invalid' in 'autocommit = invalid'. "
+                "To use 'str' value, enclose it with ' or \"."
+            ),
+        ),
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = 1.
+""",
+            TomlDecodeError,
+            "",
+        ),
+    ],
+)
+def test_invalid_loading_toml(tmp_empty, ip, file_content, error_type, error_msg):
+    Path("pyproject.toml").write_text(file_content)
+    os.makedirs("sub")
+    os.chdir("sub")
+
+    out = ip.run_cell("%load_ext sql")
+
+    assert isinstance(out.error_in_exec, error_type)
+    assert error_msg in str(out.error_in_exec)
+
+
+@pytest.mark.parametrize(
+    "file_content, expect, confirm, revert",
+    [
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocom = true
+autop = false
+autolimit = "text"
+invalid = false
+displaycon = false
+""",
+            [
+                "Found pyproject.toml",
+                "'autocom' is an invalid configuration. Did you mean 'autocommit'?",
+                (
+                    "'autop' is an invalid configuration. "
+                    "Did you mean 'autopandas', or 'autopolars'?"
+                ),
+                (
+                    "'text' is an invalid value for 'autolimit'. "
+                    "Please use <class 'int'> value instead."
+                ),
+                (
+                    "'invalid' is an invalid configuration. Please review our "
+                    "configuration guideline: https://jupysql.ploomber.io/en/latest/api/configuration.html#options"  # noqa
+                ),
+                r"displaycon\s*\|\s*False",
+            ],
+            {"displaycon": False, "autolimit": 0},
+            {"displaycon": True},
+        ),
+    ],
+)
+def test_mixed_loading_toml(
+    tmp_empty, ip, capsys, file_content, expect, confirm, revert
+):
+    Path("pyproject.toml").write_text(file_content)
+    os.makedirs("sub")
+    os.chdir("sub")
+
+    ip.run_cell("%load_ext sql").result
+    out, _ = capsys.readouterr()
+
+    assert all(re.search(substring, out) for substring in expect)
+
+    sql = ip.find_cell_magic("sql").__self__
+    assert all([getattr(sql, config) == value for config, value in confirm.items()])
```

Comment:
Here also, please create a variable for the expected outcome and use it for assertion.
You can check any of the existing test cases like [this one](https://github.com/ploomber/jupysql/blob/436bdbd993deec3502c50ba5d32bd7bfb26c7a18/src/tests/test_connection.py#L312)

Reply:
the list of configs to check is now in `confirm` variable in a dictionary format.